tests/common/include/tests/common/tools/create_toy_geometry.hpp

```diff
@@ -5,353 +5,928 @@
-                scalar m_phi = min_phi + phiBin * phi_step;
-                m_centers.push_back(
-                    point3{m_r * std::cos(m_phi), m_r * std::sin(m_phi), m_z});
-            }
-        }
+    /** source link */
+    const dindex inv_sf_finder = dindex_invalid;
+    /** Leaving world */
+    const dindex leaving_world = dindex_invalid;
 
-        // Create surfaces
-        for (auto& m_center : m_centers) {
-
-            // Build the transform
-            // The local phi
-            scalar m_phi = algebra::getter::phi(m_center);
-            // Local z axis is the normal vector
-            vector3 m_local_z{std::cos(m_phi + m_tilt_phi),
-                              std::sin(m_phi + m_tilt_phi), 0.};
-            // Local x axis the normal to local y,z
-            vector3 m_local_x{-std::sin(m_phi + m_tilt_phi),
-                              std::cos(m_phi + m_tilt_phi), 0.};
-
-            // add transform
-            transforms[detector_t::e_rectangle2].emplace_back(
-                ctx, m_center, m_local_z, m_local_x);
-
-            // add mask
-            masks.template add_mask<detector_t::e_rectangle2>(m_half_x,
-                                                              m_half_y);
-            masks.template group<detector_t::e_rectangle2>().back().links() = {
-                volume_id, dindex_invalid};
+    // create four barrel layers with modules
+    std::size_t n_brl_layers = 4;
+    // create three endcap layers with modules
+    std::size_t n_edc_layers = 3;
+
+    //
+    // barrel
+    //
+    const scalar brl_half_z = 500.;
+    const std::vector<scalar> brl_positions = {19., 32., 72., 116., 172.};
+    const std::vector<std::pair<scalar, scalar>> brl_lay_sizes = {
+        {0., 27.}, {27., 38.}, {64., 80.}, {108., 124.}, {164., 180.}};
+    const std::vector<std::pair<int, int>> brl_binning = {
+        {0., 0.}, {16, 14}, {32, 14}, {52, 14}, {78, 14}};
+    // module parameters
+    struct brl_m_config {
+        scalar m_half_x = 8.4;
+        scalar m_half_y = 36.;
+        scalar m_tilt_phi = 0.14;  // 0.145;
+        scalar layer_r = 32.;
+        scalar m_radial_stagger = 0.5;  // 2.;
+        scalar m_long_overlap = 2.;     // 5.;
+        std::pair<int, int> m_binning = {16, 14};
+    };
 
-            // create surface
-            surface surf(
-                transforms[detector_t::e_rectangle2].size(ctx) - 1,
-                {detector_t::e_rectangle2,
-                 masks.template group<detector_t::e_rectangle2>().size() - 1},
-                volume_id, dindex_invalid);
+    //
+    // endcaps
+    //
+    const std::vector<scalar> edc_positions = {600.,  700.,  820., 960.,
+                                               1100., 1300., 1500.};
+    const std::vector<std::pair<scalar, scalar>> edc_lay_sizes = {
+        {595., 605.},   {695., 705.},   {815., 825.},  {955., 965.},
+        {1095., 1105.}, {1295., 1305.}, {1495., 1505.}};
```

Comment:
Can you confirm that it does have seven layers not three ?

Reply:
Those are all of the seven layers of the TML detector. The """n_edc_layers""" variable determines how many layers are actually built. The unit test, however, can only check the version with three endcap layers, because its is not generic, yet

Reply:
OK. I also prefer a simpler geometry. TBH, three endcaps and four barrels already look a bit too much to me.  Two endcaps at each side and two barrels might be enough to test the navigation algorithm, and keep the toy detector minimal

Reply:
Hmm, I come to think it is not bad to make toy detector same with tml detector in case we test KF with traccc data, which is based on tml detector.

Reply:
You can set ```n_edc_layers``` to two to get a smaller detector. The reason I had to add all barrel layers was to have barrel and endcaps fit smoothly. Otherwise I would have needed to play with the disc diameter...

Reply:
Update: Now the number of layers can be configured freely